interp/src/primitives/stateful.rs

```diff
@@ -66,92 +102,23 @@ impl Primitive for StdMultPipe {
         }
     }
 
-    /// Currently a 1 cycle std_mult_pipe that has a similar interface
-    /// to a register.
     fn execute(
         &mut self,
         inputs: &[(calyx::ir::Id, &Value)],
-        done_val: Option<&Value>,
     ) -> Vec<(calyx::ir::Id, crate::values::OutputValue)> {
         //unwrap the arguments, left, right, and go
         let (_, left) = inputs.iter().find(|(id, _)| id == "left").unwrap();
         let (_, right) = inputs.iter().find(|(id, _)| id == "right").unwrap();
         let (_, go) = inputs.iter().find(|(id, _)| id == "go").unwrap();
         //continue computation
         if go.as_u64() == 1 {
```

Comment:
I don't think that as written this works correctly. If the go "flickers" on for a portion of the cycle but does not remain high (i.e. is not high by the time do_tick is called) then the multiplier should not run.

It might make more sense to have the multiplier save its input ports and only perform the actual computation during the do_tick based on the incoming values. Also worth noting that if the go signal is not held high then it is not entirely clear what happens to the internal state, either it is saved but not acted on, or it is wiped altogether.

Reply:
I'm a bit confused on when that type of "flickering" could happen. Like if a multiplier's `go` port is dependent on the `out` of another stateful component? Otherwise, since all combinational primitives are evaluated to completion before the stateful primitives go, how could a `go` be high during comb. primitive evaluation and low during stateful primitive evaluation?

Reply:
Also by "save" do you just mean the `do_tick` method also accepts a vec of inputs, and just re-checks the `go`/other inputs before executing and commiting updates?